src/wm/ibus.vala

```diff
@@ -73,7 +73,7 @@ namespace Budgie {
 		* Launch the daemon as a child process so that it dies when we die
 		*/
 		private void startup_ibus() {
-			string[] cmdline = {"ibus-daemon", "--xim", "--panel", "disable"};
+			string[] cmdline = {"ibus-daemon", "--xim", "--daemonize"};
```

Comment:
Is daemonizing going to put the ugly ibus icon into the system tray permanently? There has been a back-and-forth over the years about enabling the panel option and it always comes down to a system tray item sticking out like a sore thumb.

Reply:
interesting question.  On UB we override the dconf key so it doesn't display if the system tray is used. We don't carry the system tray as part of our default layout anyway - so a bit belt and braces.  For some layouts such as Japanese, if the user is NOT using a japanese keyboard with the extra keys to change the input mode, the only way to change the input mode (at least that I can find) is to re-enable the system tray icon (via ibus-setup) and change the input mode that way.

As far as I can gather on Debian/Ubuntu, the popup is a necessity for non-latin layout users which is a  little surprising statement I suppose since I haven't seen any previous comments on non debian/ubuntu forums for the lack-of.

Why the keyboard layout applet doesnt handle input modes is another question - and a different issue.

Reply:
> We don't carry the system tray as part of our default layout anyway - so a bit belt and braces.

Sure but it *is* the default for Budgie itself, which is what should be tested against :)

> On UB we override the dconf key so it doesn't display if the system tray is used

Gotcha. Yea this sounds then like it is more something that needs to be decided on by anyone shipping Budgie Desktop (like UB does).

Reply:
Can it not be launched with both options?

Reply:
the --panel disable hides it from the system tray.  But then there is no means to re-enable it.  Certain non-latin layouts such as japanese you need a means to change the input mode (if the keyboard you are using doesnt have the input mode key) which our current keyboard applet does not support.  How to extend the applet - no idea off the top of my head.

Reply:
Could include an override in carbontray to hide the icon, with an associated applet option to unhide it.

Reply:
@serebit That is a good idea.

Reply:
Would be nice to see this get implemented as part of this PR.

Reply:
Override added and as discusses a BDS option to toggle assuming ibus-daemon is installed and running.